HW1/P3/P3.py

```diff
@@ -0,0 +1,50 @@
+import findspark
+findspark.init()
+import pyspark
+import json
+sc = pyspark.SparkContext(appName="Spark1")
+
+
+
+
+# This function produce the final result format
+# input----word: the word in the .txt file
+# ouput----a tuple in the required format
+def resultmapper(word):
+    anagrams = []
+    for i in word[1]:
+        anagrams.append(i)
+    return (''.join(sorted(word[1][0])),len(word[1]),anagrams)
+
+
+# This function maps the format for the raw RDD. Using the buildin
+# hash function to produce the key.
+# Key is the unique hash code for the string and Value is the list of the word
+
+# input----word: the word in the .txt file
+# ouput----a Key, Value tuple
+def K_V_mapper(word):
+	key = hash(''.join(sorted(word)))
+	value = [word]
+	return (key,value)
+
+# This is the function maps to reduceByKey, add up all the anagrams in one list
+def reduce_mapper(item_1, item_2):
+	return item_1 + item_2
+
+rawRDD = sc.textFile('EOWL_words.txt')
+
+pairedRDD = rawRDD.map(K_V_mapper) # each item in rawRDD is something like (hashCode, word) 
```

Comment:
Although this is probably fine, it seems complicated. Is making your own hash faster or something?